src/executables/util_sa.c

```diff
@@ -1736,11 +1738,22 @@ diagdb (UTIL_FUNCTION_ARG * arg)
 	    {
 	      if (utility_check_class_name (class_name) != NO_ERROR)
 		{
+		  need_db_shutdown = true;
```

Comment:
After a successful call of db_restart(), if branching to error_exit:, it is necessary to always call db_shutdown(). Therefore, it is more appropriate to set once need_db_shutdown to true immediately after the successful invocation of db_restart().

```
  if (db_restart (arg->command_name, TRUE, db_name) != NO_ERROR)
    {
      PRINT_AND_LOG_ERR_MSG ("%s\n", db_error_string (3));
      goto error_exit;
    }

    need_db_shutdown = true;
```

Reply:
I understand. I changed it.